queries/go/highlights.scm

```diff
@@ -5,6 +5,7 @@
 ; Identifiers
 
 (type_identifier) @type
```

Comment:
This capture is not used at the moment (despite being documented) and I don't think it makes sense to add it. There is a plan to bring back opt-in semantic highlighting but I think we shouldn't mix it with regular highlighting. Especially when we decide to have `@type.definition` then there should be probably something analogous for `@xxx.definition` for other type of definition.

In the past, we had the policy that newly introduced highlights should have their own dedicated PRs to discuss them better and also add them for multiple languages to see and discuss whether they make sense. This also makes it easier to create a tracking issue to list the remaining languages that are still missing that new highlight group

Reply:
So I'd prefer plain `@type` for now and have a discussion for `@type.definition` in a separate PR. If this highlight is important it should probably be added to the parsers in `tree-sitter/tree-sitter-Xxx` as they are the most popular languages

Reply:
I supposed you mean you would like to be consistent and also have something like `method.definition` ? 
I removed `@type.definition` from the MR.

Reply:
`@type.definition` is important just to be able to match how the other IDE work (VSCode/JetBrains). Both are able to discriminate between usage and declaration. I just looked at https://github.com/nvim-treesitter/nvim-treesitter/blob/e2b2d2357b227b1d2548b6bfb3506b3a93b9e636/lua/nvim-treesitter/highlight.lua#L113 and saw `@type.definition` and I thought it's perfectly valid to add it 😁

Reply:
I created another PR for that: https://github.com/nvim-treesitter/nvim-treesitter/pull/3154